drivers/stepper/stepper_shell.c

```diff
@@ -0,0 +1,458 @@
+	}
+
+	err = parse_device_arg(sh, argv, &dev);
+	if (err < 0) {
+		return err;
+	}
+
+	err = stepper_set_micro_step_res(dev, resolution);
+	if (err) {
+		shell_error(sh, "Error: %d", err);
+	}
+
+	return err;
+}
+
+static int cmd_stepper_get_micro_step_res(const struct shell *sh, size_t argc, char **argv)
+{
+	const struct device *dev;
+	int err;
+	enum micro_step_resolution micro_step_res = 0;
+
+	err = parse_device_arg(sh, argv, &dev);
+	if (err < 0) {
+		return err;
+	}
+
+	err = stepper_get_micro_step_res(dev, &micro_step_res);
+	if (err < 0) {
+		shell_warn(sh, "Failed to get micro-step resolution: %d", err);
+	} else {
+		shell_print(sh, "Micro-step Resolution: %d", micro_step_res);
+	}
+
+	return err;
+}
+
+static int cmd_stepper_set_actual_position(const struct shell *sh, size_t argc, char **argv)
+{
+	const struct device *dev;
+	int err;
+	int32_t position = strtol(argv[arg_idx_param], NULL, 0);
+
+	err = parse_device_arg(sh, argv, &dev);
+	if (err < 0) {
+		return err;
+	}
+
+	err = stepper_set_actual_position(dev, position);
+	if (err) {
+		shell_error(sh, "Error: %d", err);
+	}
+
+	return err;
+}
+
+static int cmd_stepper_get_actual_position(const struct shell *sh, size_t argc, char **argv)
+{
+	const struct device *dev;
+	int err;
+	int32_t actual_position;
+
+	err = parse_device_arg(sh, argv, &dev);
+	if (err < 0) {
+		return err;
+	}
+
+	err = stepper_get_actual_position(dev, &actual_position);
+	if (err < 0) {
+		shell_warn(sh, "Failed to get actual position: %d", err);
+	} else {
+		shell_print(sh, "Actual Position: %d", actual_position);
+	}
+
+	return err;
+}
+
+static int cmd_stepper_set_target_position(const struct shell *sh, size_t argc, char **argv)
+{
+	const struct device *dev;
+	int err;
+	const int32_t position = strtol(argv[arg_idx_param], NULL, 0);
+	struct k_poll_signal *poll_signal =
+		COND_CODE_1(CONFIG_STEPPER_SHELL_ASYNC, (&stepper_signal), (NULL));
+
+	err = parse_device_arg(sh, argv, &dev);
+	if (err < 0) {
+		return err;
+	}
+
+#ifdef CONFIG_STEPPER_SHELL_ASYNC
+	start_polling(sh);
+#endif /* CONFIG_STEPPER_SHELL_ASYNC */
+
+	err = stepper_set_target_position(dev, position, poll_signal);
+	if (err) {
+		shell_error(sh, "Error: %d", err);
+	}
+
+	return err;
+}
+
+static int cmd_stepper_enable_constant_velocity_mode(const struct shell *sh, size_t argc,
+						     char **argv)
+{
+	const struct device *dev;
+	int err;
+	enum stepper_direction direction = {-EINVAL};
+
+	for (int i = 0; i < ARRAY_SIZE(stepper_direction_map); i++) {
+		if (strcmp(argv[arg_idx_param], stepper_direction_map[i].name) == 0) {
+			direction = stepper_direction_map[i].direction;
+		}
+	}
+	if (direction == -EINVAL) {
+		return -EINVAL;
+	}
+	uint32_t velocity = strtoul(argv[arg_idx_value], NULL, 0);
+
+	err = parse_device_arg(sh, argv, &dev);
+	if (err < 0) {
+		return err;
+	}
+
+	err = stepper_enable_constant_velocity_mode(dev, direction, velocity);
+	if (err) {
+		shell_error(sh, "Error: %d", err);
+	}
+
+	return err;
```

Comment:
`return err` above and `return 0` here, less likely to introduce a bug if you add stuff in between, same in the other functions